### ISSUE TITLE
statsd-emitter should transform dots in hostname

### ISSUE THREAD

**Issue by Erbureth** [apache/incubator-druid#4330]
When statsd-emitter adds hostname of a druid node to the metric name, it usually contains dots, which makes it difficult to parse with subsequent tools. E. g.:
```
druidcluster.druidworker1.example.com.8082.druidb.broker.query.node.ttfb.druidworker1.example.com.8083:241
druidsubcluster.druidworker1.subcluster.example.com.8082.druidb.broker.query.node.ttfb.druidworker1.subcluster.example.com.8083:241
```
If the dots were replaced by underscores, the hostname would be seen as one field and easy to single out, i. e. both

```
druidcluster.druidworker1_example_com.8082.druidb.broker.query.node.ttfb.druidworker1_example.com_8083:241
druidsubcluster.druidworker1_subcluster_example_com.8082.druidb.broker.query.node.ttfb.druidworker1_subcluster_example_com.8083:241

```
would be matched by
```
*.*.*.*.query.node.ttfb.*.*
```

**Comment by sascha-coenen** [comment]
hah, thanks for writing this. When the statsd-emitter got released I bumped into this as well and modified the code to replace dots in metric names with underscores as you suggest. Without it, host  names would turn into whole metric paths.
However, I thought that maybe I just wasn't getting how statsd works because the emitter has a way to customize the statsd metric separator. Don't know what it is good for. I tried it out but with no success.

I would like it if people agreed to add this to the upstream statsd emitter. Then I could finally retire our custom implementation and use the upstream code.
